utils/dataExport.js

```diff
@@ -0,0 +1,22 @@
+import axios from 'axios'
+
+export const exportToCsv = async (query) => {
+  const qs = new URLSearchParams(query).toString()
+  const reqUrl = `${process.env.NEXT_PUBLIC_AGGREGATION_API || process.env.NEXT_PUBLIC_MEASUREMENTS_URL}/api/v1/aggregation?${qs}&format=CSV`
+  try {
+    const response = await axios.get(reqUrl)
+    const element = document.createElement('a')
+    element.href = 'data:text/csv;charset=utf-8,' + encodeURIComponent(response.data)
```

Comment:
Since these files can become quite large I suggest we don't encode them using dataURI, but rather just provide a direct link to the CSV file. This has the added benefit that a user will be able to right click on the link and copy it to their clip board for download.

I also doesn't involve having to download the file twice in the browser.

Reply:
I suggest also setting the `download="${YY-MM-DD}-ooni-data.csv` attribute so that when the user clicks on the link, it will trigger a download of it instead of opening it in the view.

Reply:
Nevermind the above comment, I got it working.